Model/Api/AdyenGiftcard.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ *
+ * Adyen Payment Module
+ *
+ * Copyright (c) 2023 Adyen N.V.
+ * This file is open source and available under the MIT license.
+ * See the LICENSE file for more info.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Model\Api;
+
+use Adyen\Payment\Api\AdyenGiftcardInterface;
+use Adyen\Payment\Helper\Data;
+use Adyen\Payment\Model\ResourceModel\StateData\Collection as StateDataCollection;
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\Quote\Api\CartRepositoryInterface;
+use Magento\Quote\Api\Data\CartInterface;
+
+class AdyenGiftcard implements AdyenGiftcardInterface
+{
+    private StateDataCollection $adyenStateData;
+    private CartRepositoryInterface $quoteRepository;
+    private Data $adyenHelper;
+
+    /**
+     * @param StateDataCollection $adyenStateData
+     * @param CartRepositoryInterface $quoteRepository
+     * @param Data $adyenHelper
+     */
```

Comment:
any reason to have those attributes? 
I think since we have the function input parameters are typed, maybe they are no longer needed